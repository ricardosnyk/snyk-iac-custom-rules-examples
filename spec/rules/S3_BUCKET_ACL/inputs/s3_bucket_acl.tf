# Â© 2023 Snyk Limited All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

provider "aws" {
  region = "us-east-1"
}

resource "aws_s3_bucket" "valid" {
  bucket = "my-tf-valid-bucket"
}

resource "aws_s3_bucket_ownership_controls" "valid" {
  bucket = aws_s3_bucket.valid.id
  rule {
    object_ownership = "BucketOwnerPreferred"
  }
}

resource "aws_s3_bucket_acl" "valid" {
  depends_on = [aws_s3_bucket_ownership_controls.valid]

  bucket = aws_s3_bucket.valid.id
  acl    = "private"
}

resource "aws_s3_bucket" "invalid" {
  bucket = "my-invalid-tf-bucket"
}

resource "aws_s3_bucket_ownership_controls" "invalid" {
  bucket = aws_s3_bucket.invalid.id
  rule {
    object_ownership = "BucketOwnerPreferred"
  }
}

resource "aws_s3_bucket_public_access_block" "invalid" {
  bucket = aws_s3_bucket.invalid.id

  block_public_acls       = false
  block_public_policy     = false
  ignore_public_acls      = false
  restrict_public_buckets = false
}

resource "aws_s3_bucket_acl" "invalid" {
  depends_on = [
    aws_s3_bucket_ownership_controls.invalid,
    aws_s3_bucket_public_access_block.invalid,
  ]

  bucket = aws_s3_bucket.invalid.id
  acl    = "public-read"
}