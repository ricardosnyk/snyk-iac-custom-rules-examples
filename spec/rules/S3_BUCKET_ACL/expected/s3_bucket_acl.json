[
  {
    "passed": false,
    "ignored": false,
    "resource_id": "aws_s3_bucket.invalid",
    "resource_namespace": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
    "resource_type": "aws_s3_bucket",
    "severity": "critical",
    "resources": [
      {
        "id": "aws_s3_bucket.invalid",
        "type": "aws_s3_bucket",
        "namespace": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
        "location": [
          {
            "filepath": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
            "line": 37,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "bucket"
            ],
            "location": {
              "filepath": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
              "line": 38,
              "column": 3
            }
          }
        ]
      },
      {
        "id": "aws_s3_bucket_acl.invalid",
        "type": "aws_s3_bucket_acl",
        "namespace": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
        "location": [
          {
            "filepath": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
            "line": 57,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "acl"
            ],
            "location": {
              "filepath": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
              "line": 64,
              "column": 3
            }
          },
          {
            "path": [
              "bucket"
            ],
            "location": {
              "filepath": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
              "line": 63,
              "column": 3
            }
          }
        ]
      }
    ]
  },
  {
    "passed": true,
    "ignored": false,
    "resource_id": "aws_s3_bucket.valid",
    "resource_namespace": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
    "resource_type": "aws_s3_bucket",
    "severity": "critical",
    "resources": [
      {
        "id": "aws_s3_bucket.valid",
        "type": "aws_s3_bucket",
        "namespace": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
        "location": [
          {
            "filepath": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
            "line": 19,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "bucket"
            ],
            "location": {
              "filepath": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
              "line": 20,
              "column": 3
            }
          }
        ]
      },
      {
        "id": "aws_s3_bucket_acl.valid",
        "type": "aws_s3_bucket_acl",
        "namespace": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
        "location": [
          {
            "filepath": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
            "line": 30,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "acl"
            ],
            "location": {
              "filepath": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
              "line": 34,
              "column": 3
            }
          },
          {
            "path": [
              "bucket"
            ],
            "location": {
              "filepath": "spec/rules/S3_BUCKET_ACL/inputs/s3_bucket_acl.tf",
              "line": 33,
              "column": 3
            }
          }
        ]
      }
    ]
  }
]